src/test/java/uk/gov/pay/directdebit/webhook/gocardless/services/WebhookGoCardlessServiceTest.java

```diff
@@ -0,0 +1,122 @@
+        verify(mockedGoCardlessEventDao).updatePaymentRequestEventId(goCardlessEvent.getId(), paymentRequestEvent.getId());
+    }
+
+    @Test
+    public void shouldNotHandleMandateEventsWithAnInvalidResource() {
+        GoCardlessEvent goCardlessEvent = aGoCardlessEventFixture().withResourceType("not_handled").withAction("created").toEntity();
+
+        List<GoCardlessEvent> events = Collections.singletonList(goCardlessEvent);
+        webhookGoCardlessService.handleEvents(events);
+
+        verify(mockedGoCardlessEventDao).insert(goCardlessEvent);
+        verifyZeroInteractions(mockedGoCardlessService);
+        verifyZeroInteractions(mockedTransactionService);
+    }
+
+    @Test
+    public void shouldNotHandlePaymentEventsWithAnInvalidAction() {
+        GoCardlessEvent goCardlessEvent = aGoCardlessEventFixture().withResourceType("payments").withAction("not_handled").toEntity();
+
+        List<GoCardlessEvent> events = Collections.singletonList(goCardlessEvent);
+        webhookGoCardlessService.handleEvents(events);
+        verify(mockedGoCardlessEventDao).insert(goCardlessEvent);
+        verifyZeroInteractions(mockedGoCardlessService);
+        verifyZeroInteractions(mockedTransactionService);
+    }
+
+    @Test
+    public void shouldNotHandleMandateEventsWithAnInvalidAction() {
+        GoCardlessEvent goCardlessEvent = aGoCardlessEventFixture().withResourceType("mandates").withAction("not_handled").toEntity();
+
+        List<GoCardlessEvent> events = Collections.singletonList(goCardlessEvent);
+        webhookGoCardlessService.handleEvents(events);
+        verify(mockedGoCardlessEventDao).insert(goCardlessEvent);
+        verifyZeroInteractions(mockedGoCardlessService);
+        verifyZeroInteractions(mockedTransactionService);
+    }
+    @Test
+    public void shouldHandleMandateEventsWithAValidAction() {
+        GoCardlessEvent goCardlessEvent = aGoCardlessEventFixture().withResourceType("mandates").withAction("created").toEntity();
+        GoCardlessMandate goCardlessMandate = aGoCardlessMandateFixture().toEntity();
+        Transaction transaction = aTransactionFixture().toEntity();
+        PaymentRequestEvent paymentRequestEvent = PaymentRequestEvent.mandateCreated(transaction.getPaymentRequestId());
+        when(mockedGoCardlessService.findMandateForEvent(goCardlessEvent)).thenReturn(goCardlessMandate);
+        when(mockedTransactionService.findChargeForMandateId(goCardlessMandate.getMandateId())).thenReturn(transaction);
+        when(mockedTransactionService.mandateCreatedFor(transaction)).thenReturn(paymentRequestEvent);
+
+        List<GoCardlessEvent> events = Collections.singletonList(goCardlessEvent);
+        webhookGoCardlessService.handleEvents(events);
+        verify(mockedGoCardlessEventDao).insert(goCardlessEvent);
+        verify(mockedGoCardlessEventDao).updatePaymentRequestEventId(goCardlessEvent.getId(), paymentRequestEvent.getId());
+    }
+}
```

Comment:
Can we reorder the tests here? They jump around from payments to mandates and valid and invalid a bit